sleap_nn/config/data_config.py

```diff
@@ -0,0 +1,166 @@
+        max_height: (int) Maximum height the image should be padded to. If not provided, the original image size will be retained. Default: None.
+        max_width: (int) Maximum width the image should be padded to. If not provided, the original image size will be retained. Default: None.
+        scale: (float or List[float]) Factor to resize the image dimensions by, specified as either a float scalar or as a 2-tuple of [scale_x, scale_y]. If a scalar is provided, both dimensions are resized by the same factor.
+        crop_hw: (Tuple[int]) Crop height and width of each instance (h, w) for centered-instance model. If None, this would be automatically computed based on the largest instance in the sio.Labels file.
+        min_crop_size: (int) Minimum crop size to be used if crop_hw is None.
+    """
+
+    is_rgb: bool = True
+    max_height: Optional[int] = None
+    max_width: Optional[int] = None
+    scale: Any = 1.0
+    crop_hw: Optional[Tuple[int, int]] = None
+    min_crop_size: int = 32  # to help app work incase of error
+
+    def __attrs_post_init__(self):
+        self.validate_scale()
+
+    def validate_scale(self):
+        if isinstance(self.scale, float) and self.scale >= 0:
+            return
+        if isinstance(self.scale, list) and all(
+            isinstance(x, float) and x >= 0 for x in self.scale
+        ):
+            return
+        raise ValueError(
+            "PreprocessingConfig's scale must be a float or a list of floats."
+        )
+
+
+def validate_proportion(instance, attribute, value):
+    if not (0.0 <= value <= 1.0):
+        raise ValueError(f"{attribute.name} must be between 0.0 and 1.0, got {value}")
+
+
+@attrs.define
+class IntensityConfig:
+    """Configuration of Intensity (Optional):
+
+    Attributes:
+        uniform_noise_min: (float) Minimum value for uniform noise (uniform_noise_min >=0).
+        uniform_noise_max: (float) Maximum value for uniform noise (uniform_noise_max <>=1).
+        uniform_noise_p: (float) Probability of applying random uniform noise. Default=0.0
+        gaussian_noise_mean: (float) The mean of the gaussian noise distribution.
+        gaussian_noise_std: (float) The standard deviation of the gaussian noise distribution.
+        gaussian_noise_p: (float) Probability of applying random gaussian noise. Default=0.0
+        contrast_min: (float) Minimum contrast factor to apply. Default: 0.5.
+        contrast_max: (float) Maximum contrast factor to apply. Default: 2.0.
+        contrast_p: (float) Probability of applying random contrast. Default=0.0
+        brightness: (float) The brightness factor to apply. Default: (1.0, 1.0).
+        brightness_p: (float) Probability of applying random brightness. Default=0.0
+    """
+
+    uniform_noise_min: float = attrs.field(
+        default=0.0, validator=attrs.validators.ge(0)
+    )
```

Comment:
Import field directly from attrs and use
```suggestion
    uniform_noise_min: float = field(
        default=0.0, validator=attrs.validators.ge(0)
    )
```